apps/src/codebridge/ControlButtons/index.tsx

```diff
@@ -53,22 +59,49 @@ const ControlButtons: React.FunctionComponent = () => {
   // No-op for now. TODO: figure out what the finish button should do.
   // https://codedotorg.atlassian.net/browse/CT-664
   const onFinish = () => {};
-  const onSubmit = () =>
+
+  const onSubmit = () => {
+    const dialogTitle = hasSubmitted
+      ? commonI18n.unsubmitYourProject()
+      : commonI18n.submitYourProject();
+    const dialogMessage = hasSubmitted
+      ? commonI18n.unsubmitYourProjectConfirm()
+      : commonI18n.submitYourProjectConfirm();
+    dialogControl?.showDialog(
+      DialogType.GenericConfirmation,
+      handleSubmit,
+      dialogTitle,
+      dialogMessage
+    );
+  };
+
+  const handleSubmit = () => {
     dispatch(
       sendSubmitReport({appType: appType || '', submitted: !hasSubmitted})
     );
+    // Go to the next level if we have one and we just submitted.
+    // TODO: If onContinue starts to update progress, make sure we don't override the submitted status.
```

Comment:
Do we need to file a ticket for this? Who would determine whether `onContinue` starts to update progress?

Reply:
Good call out! We already have a ticket to cover updating progress appropriately: https://codedotorg.atlassian.net/browse/CT-526, I added a note to make sure we don't break submittable levels.